modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteTypeCoercion.java

```diff
@@ -91,6 +95,34 @@ public boolean binaryComparisonCoercion(SqlCallBinding binding) {
         }
     }
 
+    static @Nullable SqlLiteral extractLiteral(SqlCall call, SqlValidator validator) {
+        if (call.getOperator().getKind() == SqlKind.CAST) {
```

Comment:
I think it is better to move this method to the validator.

Reply:
done

Reply:
now it used only in one place, thus i store it as is.